// wp-e2e-page-objects/lib/pages/page.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */


var _seleniumWebdriver = require('selenium-webdriver');

var _lodash = require('lodash');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var defaultArgs = {
	url: '/',
	visit: true,
	waitMs: 10000,
	components: {}
};

var Page = function () {
	function Page(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, Page);

		this.driver = driver;

		args = Object.assign(defaultArgs, args);

		this.url = args.url;
		this.waitMs = args.waitMs;

		if (args.visit) {
			this.visit();
		}

		var pageComponents = {};
		(0, _lodash.forEach)(args.components, function (ClassName, name) {
			pageComponents[name] = new ClassName(driver);
		});

		this.components = pageComponents;
	}

	_createClass(Page, [{
		key: 'getCurrentUrl',
		value: function getCurrentUrl() {
			return this.driver.getCurrentUrl();
		}
	}, {
		key: 'getTitle',
		value: function getTitle() {
			return this.driver.getTitle();
		}
	}, {
		key: 'titleContains',
		value: function titleContains(pattern) {
			return this.getTitle().then(function (title) {
				var re = new RegExp(pattern);
				return re.test(title);
			}, function () {
				return false;
			});
		}
	}, {
		key: 'hasText',
		value: function hasText(text) {
			var selector = _seleniumWebdriver.By.xpath('//body//*[contains(., "' + text + '")]');
			return this.driver.findElement(selector).then(function (el) {
				return el.isDisplayed();
			}, function () {
				return false;
			});
		}
	}, {
		key: 'visit',
		value: function visit() {
			this.driver.get(this.url);
		}
	}]);

	return Page;
}();

exports.default = Page;
module.exports = exports['default'];
