// wp-e2e-page-objects/lib/components/component-comment.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _component = require('./component');

var _component2 = _interopRequireDefault(_component);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var CONTAINER_SELECTOR = _seleniumWebdriver.By.css('#respond');
var COMMENT_SELECTOR = _seleniumWebdriver.By.css('#comment');
var NAME_SELECTOR = _seleniumWebdriver.By.css('#author');
var EMAIL_SELECTOR = _seleniumWebdriver.By.css('#email');
var URL_SELECTOR = _seleniumWebdriver.By.css('#url');
var SUBMIT_SELECTOR = _seleniumWebdriver.By.css('[name="submit"]');

var ComponentComment = function (_Component) {
	_inherits(ComponentComment, _Component);

	function ComponentComment(driver) {
		var selector = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : CONTAINER_SELECTOR;

		_classCallCheck(this, ComponentComment);

		return _possibleConstructorReturn(this, (ComponentComment.__proto__ || Object.getPrototypeOf(ComponentComment)).call(this, driver, selector));
	}

	_createClass(ComponentComment, [{
		key: 'setComment',
		value: function setComment(comment) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, COMMENT_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, COMMENT_SELECTOR, comment);
		}
	}, {
		key: 'setName',
		value: function setName(name) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, NAME_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, NAME_SELECTOR, name);
		}
	}, {
		key: 'setEmail',
		value: function setEmail(email) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveto(this.driver, EMAIL_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, EMAIL_SELECTOR, email);
		}
	}, {
		key: 'setUrl',
		value: function setUrl(url) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, URL_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, URL_SELECTOR, url);
		}
	}, {
		key: 'submit',
		value: function submit() {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, SUBMIT_SELECTOR);

			// Even after mouseMoveTo, it still not clickable. This seems hacky,
			// but it works by scrolling the page down.
			_wpE2eWebdriver.WebDriverHelper.scrollDown(this.driver);

			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, SUBMIT_SELECTOR);
		}
	}]);

	return ComponentComment;
}(_component2.default);

exports.default = ComponentComment;
module.exports = exports['default'];
