// wp-e2e-page-objects/lib/components/component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */


var _wpE2eWebdriver = require('wp-e2e-webdriver');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Component = function () {
	function Component(driver, selector) {
		var args = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : { wait: true, waitMs: 10000 };

		_classCallCheck(this, Component);

		this.driver = driver;
		this.selector = selector;
		this.waitMs = args.waitMs;

		if (this.wait) {
			this.waitElement();
		}
	}

	_createClass(Component, [{
		key: 'waitElement',
		value: function waitElement() {
			return _wpE2eWebdriver.WebDriverHelper.waitTillPresentAndDisplayed(this.driver, this.selector, this.waitMs);
		}
	}, {
		key: 'displayed',
		value: function displayed() {
			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, this.selector, this.waitMs);
		}
	}]);

	return Component;
}();

exports.default = Component;
module.exports = exports['default'];
