// wc-e2e-page-objects/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.CustomerFlow = exports.GuestCustomerFlow = exports.StoreOwnerFlow = exports.WPAdminWCSettingsProductsGeneral = exports.WPAdminWCSettingsProductsDownloadable = exports.WPAdminWCSettingsTaxRates = exports.WPAdminWCSettingsTax = exports.WPAdminWCSettingsGeneral = exports.WPAdminWCSettings = exports.WPAdminProductNew = exports.ShopPage = exports.MyAccountPage = exports.CheckoutOrderReceivedPage = exports.CheckoutPage = exports.CartPage = exports.SingleProductPage = exports.ComponentOrderShippingAddress = exports.ComponentOrderBillingAddress = exports.ComponentOrderCustomerDetails = exports.ComponentOrderDetails = exports.ComponentCheckoutShippingDetails = exports.ComponentCheckoutOrderReview = exports.ComponentCheckoutBillingDetails = exports.ComponentCheckout = exports.Helper = exports.PageMap = undefined;

var _pageMap = require('./page-map');

var PageMap = _interopRequireWildcard(_pageMap);

var _helper = require('./helper');

var Helper = _interopRequireWildcard(_helper);

var _componentCheckout = require('./components/component-checkout');

var _componentCheckout2 = _interopRequireDefault(_componentCheckout);

var _componentCheckoutBillingDetails = require('./components/component-checkout-billing-details');

var _componentCheckoutBillingDetails2 = _interopRequireDefault(_componentCheckoutBillingDetails);

var _componentCheckoutOrderReview = require('./components/component-checkout-order-review');

var _componentCheckoutOrderReview2 = _interopRequireDefault(_componentCheckoutOrderReview);

var _componentCheckoutShippingDetails = require('./components/component-checkout-shipping-details');

var _componentCheckoutShippingDetails2 = _interopRequireDefault(_componentCheckoutShippingDetails);

var _componentOrderDetails = require('./components/component-order-details');

var _componentOrderDetails2 = _interopRequireDefault(_componentOrderDetails);

var _componentOrderCustomerDetails = require('./components/component-order-customer-details');

var _componentOrderCustomerDetails2 = _interopRequireDefault(_componentOrderCustomerDetails);

var _componentOrderBillingAddress = require('./components/component-order-billing-address');

var _componentOrderBillingAddress2 = _interopRequireDefault(_componentOrderBillingAddress);

var _componentOrderShippingAddress = require('./components/component-order-shipping-address');

var _componentOrderShippingAddress2 = _interopRequireDefault(_componentOrderShippingAddress);

var _singleProductPage = require('./pages/single-product-page.js');

var _singleProductPage2 = _interopRequireDefault(_singleProductPage);

var _cartPage = require('./pages/cart-page');

var _cartPage2 = _interopRequireDefault(_cartPage);

var _checkoutPage = require('./pages/checkout-page');

var _checkoutPage2 = _interopRequireDefault(_checkoutPage);

var _checkoutOrderReceivedPage = require('./pages/checkout-order-received-page');

var _checkoutOrderReceivedPage2 = _interopRequireDefault(_checkoutOrderReceivedPage);

var _myAccountPage = require('./pages/my-account-page');

var _myAccountPage2 = _interopRequireDefault(_myAccountPage);

var _shopPage = require('./pages/shop-page');

var _shopPage2 = _interopRequireDefault(_shopPage);

var _wpAdminProductNew = require('./pages/wp-admin/wp-admin-product-new');

var _wpAdminProductNew2 = _interopRequireDefault(_wpAdminProductNew);

var _wpAdminWcSettings = require('./pages/wp-admin/wp-admin-wc-settings');

var _wpAdminWcSettings2 = _interopRequireDefault(_wpAdminWcSettings);

var _wpAdminWcSettingsGeneral = require('./pages/wp-admin/wp-admin-wc-settings-general');

var _wpAdminWcSettingsGeneral2 = _interopRequireDefault(_wpAdminWcSettingsGeneral);

var _wpAdminWcSettingsTax = require('./pages/wp-admin/wp-admin-wc-settings-tax');

var _wpAdminWcSettingsTax2 = _interopRequireDefault(_wpAdminWcSettingsTax);

var _wpAdminWcSettingsTaxRates = require('./pages/wp-admin/wp-admin-wc-settings-tax-rates');

var _wpAdminWcSettingsTaxRates2 = _interopRequireDefault(_wpAdminWcSettingsTaxRates);

var _wpAdminWcSettingsProductsGeneral = require('./pages/wp-admin/wp-admin-wc-settings-products-general');

var _wpAdminWcSettingsProductsGeneral2 = _interopRequireDefault(_wpAdminWcSettingsProductsGeneral);

var _wpAdminWcSettingsProductsDownloadable = require('./pages/wp-admin/wp-admin-wc-settings-products-downloadable');

var _wpAdminWcSettingsProductsDownloadable2 = _interopRequireDefault(_wpAdminWcSettingsProductsDownloadable);

var _storeOwnerFlow = require('./flows/store-owner-flow');

var _storeOwnerFlow2 = _interopRequireDefault(_storeOwnerFlow);

var _guestCustomerFlow = require('./flows/guest-customer-flow');

var _guestCustomerFlow2 = _interopRequireDefault(_guestCustomerFlow);

var _customerFlow = require('./flows/customer-flow');

var _customerFlow2 = _interopRequireDefault(_customerFlow);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

/**
 * Internal dependencies
 */
exports.PageMap = PageMap;
exports.Helper = Helper;
exports.ComponentCheckout = _componentCheckout2.default;
exports.ComponentCheckoutBillingDetails = _componentCheckoutBillingDetails2.default;
exports.ComponentCheckoutOrderReview = _componentCheckoutOrderReview2.default;
exports.ComponentCheckoutShippingDetails = _componentCheckoutShippingDetails2.default;
exports.ComponentOrderDetails = _componentOrderDetails2.default;
exports.ComponentOrderCustomerDetails = _componentOrderCustomerDetails2.default;
exports.ComponentOrderBillingAddress = _componentOrderBillingAddress2.default;
exports.ComponentOrderShippingAddress = _componentOrderShippingAddress2.default;
exports.SingleProductPage = _singleProductPage2.default;
exports.CartPage = _cartPage2.default;
exports.CheckoutPage = _checkoutPage2.default;
exports.CheckoutOrderReceivedPage = _checkoutOrderReceivedPage2.default;
exports.MyAccountPage = _myAccountPage2.default;
exports.ShopPage = _shopPage2.default;
exports.WPAdminProductNew = _wpAdminProductNew2.default;
exports.WPAdminWCSettings = _wpAdminWcSettings2.default;
exports.WPAdminWCSettingsGeneral = _wpAdminWcSettingsGeneral2.default;
exports.WPAdminWCSettingsTax = _wpAdminWcSettingsTax2.default;
exports.WPAdminWCSettingsTaxRates = _wpAdminWcSettingsTaxRates2.default;
exports.WPAdminWCSettingsProductsDownloadable = _wpAdminWcSettingsProductsDownloadable2.default;
exports.WPAdminWCSettingsProductsGeneral = _wpAdminWcSettingsProductsGeneral2.default;
exports.StoreOwnerFlow = _storeOwnerFlow2.default;
exports.GuestCustomerFlow = _guestCustomerFlow2.default;
exports.CustomerFlow = _customerFlow2.default;
