// wc-e2e-page-objects/lib/page-map.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.PAGE = undefined;
exports.getPageUrl = getPageUrl;

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _shopPage = require('./pages/shop-page');

var _shopPage2 = _interopRequireDefault(_shopPage);

var _cartPage = require('./pages/cart-page');

var _cartPage2 = _interopRequireDefault(_cartPage);

var _checkoutPage = require('./pages/checkout-page');

var _checkoutPage2 = _interopRequireDefault(_checkoutPage);

var _myAccountPage = require('./pages/my-account-page');

var _myAccountPage2 = _interopRequireDefault(_myAccountPage);

var _wpAdminProductEdit = require('./pages/wp-admin/wp-admin-product-edit');

var _wpAdminProductEdit2 = _interopRequireDefault(_wpAdminProductEdit);

var _wpAdminProductNew = require('./pages/wp-admin/wp-admin-product-new');

var _wpAdminProductNew2 = _interopRequireDefault(_wpAdminProductNew);

var _wpAdminProducts = require('./pages/wp-admin/wp-admin-products');

var _wpAdminProducts2 = _interopRequireDefault(_wpAdminProducts);

var _wpAdminOrderEdit = require('./pages/wp-admin/wp-admin-order-edit');

var _wpAdminOrderEdit2 = _interopRequireDefault(_wpAdminOrderEdit);

var _wpAdminOrderNew = require('./pages/wp-admin/wp-admin-order-new');

var _wpAdminOrderNew2 = _interopRequireDefault(_wpAdminOrderNew);

var _wpAdminOrders = require('./pages/wp-admin/wp-admin-orders');

var _wpAdminOrders2 = _interopRequireDefault(_wpAdminOrders);

var _wpAdminCouponEdit = require('./pages/wp-admin/wp-admin-coupon-edit');

var _wpAdminCouponEdit2 = _interopRequireDefault(_wpAdminCouponEdit);

var _wpAdminCouponNew = require('./pages/wp-admin/wp-admin-coupon-new');

var _wpAdminCouponNew2 = _interopRequireDefault(_wpAdminCouponNew);

var _wpAdminCoupons = require('./pages/wp-admin/wp-admin-coupons');

var _wpAdminCoupons2 = _interopRequireDefault(_wpAdminCoupons);

var _wpAdminWcSettingsTax = require('./pages/wp-admin/wp-admin-wc-settings-tax');

var _wpAdminWcSettingsTax2 = _interopRequireDefault(_wpAdminWcSettingsTax);

var _wpAdminWcSettingsGeneral = require('./pages/wp-admin/wp-admin-wc-settings-general');

var _wpAdminWcSettingsGeneral2 = _interopRequireDefault(_wpAdminWcSettingsGeneral);

var _wpAdminWcSettingsCheckout = require('./pages/wp-admin/wp-admin-wc-settings-checkout');

var _wpAdminWcSettingsCheckout2 = _interopRequireDefault(_wpAdminWcSettingsCheckout);

var _wpAdminWcSettingsCheckoutBacs = require('./pages/wp-admin/wp-admin-wc-settings-checkout-bacs');

var _wpAdminWcSettingsCheckoutBacs2 = _interopRequireDefault(_wpAdminWcSettingsCheckoutBacs);

var _wpAdminWcSettingsCheckoutCod = require('./pages/wp-admin/wp-admin-wc-settings-checkout-cod');

var _wpAdminWcSettingsCheckoutCod2 = _interopRequireDefault(_wpAdminWcSettingsCheckoutCod);

var _wpAdminWcSettingsCheckoutPaypal = require('./pages/wp-admin/wp-admin-wc-settings-checkout-paypal');

var _wpAdminWcSettingsCheckoutPaypal2 = _interopRequireDefault(_wpAdminWcSettingsCheckoutPaypal);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * External dependencies
 */
var PAGE = exports.PAGE = Object.assign(_wpE2ePageObjects.PageMap.PAGE, {
	SHOP: {
		object: _shopPage2.default,
		path: '/shop'
	},
	CART: {
		object: _cartPage2.default,
		path: '/cart'
	},
	CHECKOUT: {
		object: _checkoutPage2.default,
		path: '/checkout'
	},
	MY_ACCOUNT: {
		object: _myAccountPage2.default,
		path: '/my-account'
	},
	WP_ADMIN_PRODUCTS: {
		object: _wpAdminProducts2.default,
		path: '/wp-admin/edit.php?post_type=product'
	},
	WP_ADMIN_NEW_PRODUCT: {
		object: _wpAdminProductNew2.default,
		path: '/wp-admin/post-new.php?post_type=product'
	},
	WP_ADMIN_EDIT_PRODUCT: {
		object: _wpAdminProductEdit2.default,
		path: '/wp-admin/post.php?post=%s&action=edit'
	},
	WP_ADMIN_ORDERS: {
		object: _wpAdminOrders2.default,
		path: '/wp-admin/edit.php?post_type=shop_order'
	},
	WP_ADMIN_NEW_ORDER: {
		object: _wpAdminOrderNew2.default,
		path: '/wp-admin/post-new.php?post_type=shop_order'
	},
	WP_ADMIN_EDIT_ORDER: {
		object: _wpAdminOrderEdit2.default,
		path: '/wp-admin/post.php?post=%s&action=edit'
	},
	WP_ADMIN_COUPONS: {
		object: _wpAdminCoupons2.default,
		path: '/wp-admin/edit.php?post_type=shop_coupon'
	},
	WP_ADMIN_NEW_COUPON: {
		object: _wpAdminCouponNew2.default,
		path: '/wp-admin/post-new.php?post_type=shop_coupon'
	},
	WP_ADMIN_EDIT_COUPON: {
		object: _wpAdminCouponEdit2.default,
		path: '/wp-admin/post.php?post=%s&action=edit'
	},
	WP_ADMIN_WC_SETTINGS_GENERAL: {
		object: _wpAdminWcSettingsGeneral2.default,
		path: '/wp-admin/admin.php?page=wc-settings'
	},
	WP_ADMIN_WC_SETTINGS_TAX: {
		object: _wpAdminWcSettingsTax2.default,
		path: '/wp-admin/admin.php?page=wc-settings&tab=tax'
	},
	WP_ADMIN_WC_SETTINGS_CHECKOUT: {
		object: _wpAdminWcSettingsCheckout2.default,
		path: '/wp-admin/admin.php?page=wc-settings&tab=checkout'
	},
	WP_ADMIN_WC_SETTINGS_CHECKOUT_BACS: {
		object: _wpAdminWcSettingsCheckoutBacs2.default,
		path: '/wp-admin/admin.php?page=wc-settings&tab=checkout&section=bacs'
	},
	WP_ADMIN_WC_SETTINGS_CHECKOUT_COD: {
		object: _wpAdminWcSettingsCheckoutCod2.default,
		path: '/wp-admin/admin.php?page=wc-settings&tab=checkout&section=cod'
	},
	WP_ADMIN_WC_SETTINGS_CHECKOUT_PAYPAL: {
		object: _wpAdminWcSettingsCheckoutPaypal2.default,
		path: '/wp-admin/admin.php?page=wc-settings&tab=checkout&section=paypal'
	}
});

/**
 * Internal dependencies
 */
function getPageUrl(baseUrl, page) {
	for (var _len = arguments.length, args = Array(_len > 2 ? _len - 2 : 0), _key = 2; _key < _len; _key++) {
		args[_key - 2] = arguments[_key];
	}

	return _wpE2ePageObjects.PageMap.getPageUrl.apply(_wpE2ePageObjects.PageMap, [baseUrl, page].concat(args));
}
