// wc-e2e-page-objects/lib/components/component-checkout-billing-details.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _helper = require('../helper.js');

var wcHelper = _interopRequireWildcard(_helper);

var _componentCheckout = require('./component-checkout');

var _componentCheckout2 = _interopRequireDefault(_componentCheckout);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var SELECTOR = _seleniumWebdriver.By.css('.customer-billing-fields');
var FIRST_NAME_SELECTOR = _seleniumWebdriver.By.css('#billing_first_name');
var LAST_NAME_SELECTOR = _seleniumWebdriver.By.css('#billing_last_name');
var COMPANY_SELECTOR = _seleniumWebdriver.By.css('#billing_company');
var EMAIL_SELECTOR = _seleniumWebdriver.By.css('#billing_email');
var PHONE_SELECTOR = _seleniumWebdriver.By.css('#billing_phone');
var COUNTRY_SELECTOR = _seleniumWebdriver.By.css('#select2-billing_country-container');
var ADDRESS_1_SELECTOR = _seleniumWebdriver.By.css('#billing_address_1');
var ADDRESS_2_SELECTOR = _seleniumWebdriver.By.css('#billing_address_2');
var CITY_SELECTOR = _seleniumWebdriver.By.css('#billing_city');
var STATE_SELECTOR = _seleniumWebdriver.By.css('#billing_state_field .select2-selection__arrow');
var ZIP_SELECTOR = _seleniumWebdriver.By.css('#billing_postcode');

var ComponentCheckoutBillingDetails = function (_ComponentCheckout) {
	_inherits(ComponentCheckoutBillingDetails, _ComponentCheckout);

	function ComponentCheckoutBillingDetails(driver) {
		_classCallCheck(this, ComponentCheckoutBillingDetails);

		return _possibleConstructorReturn(this, (ComponentCheckoutBillingDetails.__proto__ || Object.getPrototypeOf(ComponentCheckoutBillingDetails)).call(this, driver, SELECTOR, { wait: false }));
	}

	_createClass(ComponentCheckoutBillingDetails, [{
		key: 'setFirstName',
		value: function setFirstName(firstName) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, FIRST_NAME_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, FIRST_NAME_SELECTOR, firstName);
		}
	}, {
		key: 'setLastName',
		value: function setLastName(lastName) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, LAST_NAME_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, LAST_NAME_SELECTOR, lastName);
		}
	}, {
		key: 'setCompany',
		value: function setCompany(company) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, COMPANY_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, COMPANY_SELECTOR, company);
		}
	}, {
		key: 'setEmail',
		value: function setEmail(email) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, EMAIL_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, EMAIL_SELECTOR, email);
		}
	}, {
		key: 'setPhone',
		value: function setPhone(phone) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, PHONE_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, PHONE_SELECTOR, phone);
		}
	}, {
		key: 'selectCountry',
		value: function selectCountry(keyword, option) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, COUNTRY_SELECTOR);
			return wcHelper.select2OptionWithSearch(this.driver, COUNTRY_SELECTOR, keyword, option);
		}
	}, {
		key: 'setAddress1',
		value: function setAddress1(addr) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, ADDRESS_1_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ADDRESS_1_SELECTOR, addr);
		}
	}, {
		key: 'setAddress2',
		value: function setAddress2(addr) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, ADDRESS_2_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ADDRESS_2_SELECTOR, addr);
		}
	}, {
		key: 'setCity',
		value: function setCity(city) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, CITY_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, CITY_SELECTOR, city);
		}
	}, {
		key: 'selectState',
		value: function selectState(keyword, option) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, STATE_SELECTOR);
			return wcHelper.select2OptionWithSearch(this.driver, STATE_SELECTOR, keyword, option);
		}
	}, {
		key: 'setZip',
		value: function setZip(zip) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, ZIP_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ZIP_SELECTOR, zip);
		}
	}]);

	return ComponentCheckoutBillingDetails;
}(_componentCheckout2.default);

exports.default = ComponentCheckoutBillingDetails;
module.exports = exports['default'];
