// wc-e2e-page-objects/lib/components/component-checkout-order-review.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _componentCheckout = require('./component-checkout');

var _componentCheckout2 = _interopRequireDefault(_componentCheckout);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var SELECTOR = _seleniumWebdriver.By.css('#order_review');

var ComponentCheckoutOrderReview = function (_ComponentCheckout) {
	_inherits(ComponentCheckoutOrderReview, _ComponentCheckout);

	function ComponentCheckoutOrderReview(driver) {
		_classCallCheck(this, ComponentCheckoutOrderReview);

		return _possibleConstructorReturn(this, (ComponentCheckoutOrderReview.__proto__ || Object.getPrototypeOf(ComponentCheckoutOrderReview)).call(this, driver, SELECTOR));
	}

	_createClass(ComponentCheckoutOrderReview, [{
		key: 'hasItem',
		value: function hasItem(item, _ref) {
			var _ref$qty = _ref.qty,
			    qty = _ref$qty === undefined ? '1' : _ref$qty,
			    _ref$total = _ref.total,
			    total = _ref$total === undefined ? '$0' : _ref$total;

			var selector = _seleniumWebdriver.By.xpath('//tr[@class="cart_item" and ' + ('.//td[contains(., "' + item + '") and contains(., "\xD7 ' + qty + '")] and ') + ('.//td[contains(., "' + total + '")]') + ']');

			return _wpE2eWebdriver.WebDriverHelper.waitTillPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'hasSubtotal',
		value: function hasSubtotal(subtotal) {
			var selector = _seleniumWebdriver.By.xpath('//tr[contains(@class, "cart-subtotal") and .//td[contains(., "' + subtotal + '")]]');
			this._mouseMoveTo(selector);

			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'hasShipping',
		value: function hasShipping(shipping) {
			var selector = _seleniumWebdriver.By.xpath('//tr[contains(@class, "shipping") and .//td[contains(., "' + shipping + '")]]');
			this._mouseMoveTo(selector);

			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'hasTax',
		value: function hasTax(tax) {
			var selector = _seleniumWebdriver.By.xpath('//tr[contains(@class, "tax-rate") and .//td[contains(., "' + tax + '")]]');
			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'hasTotal',
		value: function hasTotal(total) {
			var selector = _seleniumWebdriver.By.xpath('//tr[contains(@class, "order-total") and .//td[contains(., "' + total + '")]]');
			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}]);

	return ComponentCheckoutOrderReview;
}(_componentCheckout2.default);

exports.default = ComponentCheckoutOrderReview;
module.exports = exports['default'];
