// wc-e2e-page-objects/lib/components/component-cart-item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


function getItemSelector(productTitle, args) {
	return _seleniumWebdriver.By.xpath('//tr[contains(@class, "cart_item") and ' + getProductColumnExpression(productTitle) + ' and ' + getQtyColumnExpression(args) + ']');
}

function getProductColumnExpression(productTitle) {
	return 'td[@class="product-name" and ' + ('a[contains(text(), "' + productTitle + '")]') + ']';
}

function getQtyColumnExpression(args) {
	return 'td[@class="product-quantity" and ' + './/' + getQtyInputExpression(args) + ']';
}

function getQtyInputExpression(args) {
	var qtyValue = '';

	if (args.checkQty) {
		qtyValue = ' and @value="' + args.qty + '"';
	}

	return 'input[contains(@class, "input-text")' + qtyValue + ']';
}

function getRemoveExpression() {
	return 'td[@class="product-remove"]//a[@class="remove"]';
}

var defaultArgs = {
	checkQty: true,
	qty: 1
};

var ComponentCartItem = function (_Component) {
	_inherits(ComponentCartItem, _Component);

	function ComponentCartItem(driver, productTitle) {
		var args = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};

		_classCallCheck(this, ComponentCartItem);

		args = Object.assign(defaultArgs, args);
		var selector = getItemSelector(productTitle, args);

		var _this = _possibleConstructorReturn(this, (ComponentCartItem.__proto__ || Object.getPrototypeOf(ComponentCartItem)).call(this, driver, selector, args));

		_this.productTitle = productTitle;
		_this.selector = selector;
		return _this;
	}

	_createClass(ComponentCartItem, [{
		key: 'setQty',
		value: function setQty(qty) {
			var selector = _seleniumWebdriver.By.xpath(getItemSelector(this.productTitle, { checkQty: false }).value + '//' + getQtyInputExpression({ checkQty: false }));

			var el = this.driver.findElement(selector);

			_wpE2eWebdriver.WebDriverHelper.waitForFieldClearable(this.driver, selector);
			el.sendKeys(qty);
		}
	}, {
		key: 'remove',
		value: function remove() {
			var selector = _seleniumWebdriver.By.xpath(getItemSelector(this.productTitle, { checkQty: false }).value + '//' + getRemoveExpression());

			_wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, selector);

			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, _seleniumWebdriver.By.xpath('div[@class="woocommerce-message" and contains(text(), "removed.")]'));
		}
	}]);

	return ComponentCartItem;
}(_wpE2ePageObjects.Component);

exports.default = ComponentCartItem;
module.exports = exports['default'];
