// wc-e2e-page-objects/lib/components/wp-admin/component-coupon-data-panel-general.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var PANEL_SELECTOR = _seleniumWebdriver.By.css('#general_coupon_data');
var DISCOUNT_TYPE_SELECTOR = _seleniumWebdriver.By.css('#discount_type');
var COUPON_AMOUNT_SELECTOR = _seleniumWebdriver.By.css('#coupon_amount');
var ALLOW_FREE_SHIPPING_SELECTOR = _seleniumWebdriver.By.css('#free_shipping');
var EXPIRY_DATE_SELECTOR = _seleniumWebdriver.By.css('#expiry_date');

var ComponentCouponDataPanelGeneral = function (_Component) {
	_inherits(ComponentCouponDataPanelGeneral, _Component);

	function ComponentCouponDataPanelGeneral(driver) {
		_classCallCheck(this, ComponentCouponDataPanelGeneral);

		return _possibleConstructorReturn(this, (ComponentCouponDataPanelGeneral.__proto__ || Object.getPrototypeOf(ComponentCouponDataPanelGeneral)).call(this, driver, PANEL_SELECTOR, { wait: false }));
	}

	_createClass(ComponentCouponDataPanelGeneral, [{
		key: 'selectDiscountType',
		value: function selectDiscountType(type) {
			return _wpE2eWebdriver.WebDriverHelper.selectOption(this.driver, DISCOUNT_TYPE_SELECTOR, type);
		}
	}, {
		key: 'setCouponAmount',
		value: function setCouponAmount(amount) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, COUPON_AMOUNT_SELECTOR, amount);
		}
	}, {
		key: 'checkAllowFreeShipping',
		value: function checkAllowFreeShipping() {
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, ALLOW_FREE_SHIPPING_SELECTOR);
		}
	}, {
		key: 'uncheckAllowFreeShipping',
		value: function uncheckAllowFreeShipping() {
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, ALLOW_FREE_SHIPPING_SELECTOR);
		}
	}, {
		key: 'setCouponExpiryDate',
		value: function setCouponExpiryDate(date) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, EXPIRY_DATE_SELECTOR, date);
		}
	}]);

	return ComponentCouponDataPanelGeneral;
}(_wpE2ePageObjects.Component);

exports.default = ComponentCouponDataPanelGeneral;
module.exports = exports['default'];
