// wc-e2e-page-objects/lib/components/wp-admin/component-product-data-panel-shipping.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var PANEL_SELECTOR = _seleniumWebdriver.By.css('#shipping_product_data');
var WEIGHT_SELECTOR = _seleniumWebdriver.By.css('#_weight');
var DIMENSION_LENGTH_SELECTOR = _seleniumWebdriver.By.css('input[name="_length"]');
var DIMENSION_WIDTH_SELECTOR = _seleniumWebdriver.By.css('input[name="_width"]');
var DIMENSION_HEIGHT_SELECTOR = _seleniumWebdriver.By.css('input[name="_height"]');
var SHIPPING_CLASS_SELECTOR = _seleniumWebdriver.By.css('#product_shipping_class');

var ComponentProductDataPanelShipping = function (_Component) {
	_inherits(ComponentProductDataPanelShipping, _Component);

	function ComponentProductDataPanelShipping(driver) {
		_classCallCheck(this, ComponentProductDataPanelShipping);

		return _possibleConstructorReturn(this, (ComponentProductDataPanelShipping.__proto__ || Object.getPrototypeOf(ComponentProductDataPanelShipping)).call(this, driver, PANEL_SELECTOR, { wait: false }));
	}

	_createClass(ComponentProductDataPanelShipping, [{
		key: 'setWeight',
		value: function setWeight(weight) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, WEIGHT_SELECTOR, weight);
		}
	}, {
		key: 'setDimensionLength',
		value: function setDimensionLength(length) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, DIMENSION_LENGTH_SELECTOR, length);
		}
	}, {
		key: 'setDimensionWidth',
		value: function setDimensionWidth(width) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, DIMENSION_WIDTH_SELECTOR, width);
		}
	}, {
		key: 'setDimensionHeight',
		value: function setDimensionHeight(height) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, DIMENSION_HEIGHT_SELECTOR, height);
		}
	}, {
		key: 'selectShippingClass',
		value: function selectShippingClass(option) {
			return _wpE2eWebdriver.WebDriverHelper.selectOption(this.driver, SHIPPING_CLASS_SELECTOR, option);
		}
	}]);

	return ComponentProductDataPanelShipping;
}(_wpE2ePageObjects.Component);

exports.default = ComponentProductDataPanelShipping;
module.exports = exports['default'];
