// wc-e2e-page-objects/lib/components/wp-admin/component-coupon-data-panel-usage-restriction.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _helper = require('../../helper');

var wcHelper = _interopRequireWildcard(_helper);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal helper
 */


var PANEL_SELECTOR = _seleniumWebdriver.By.css('#general_coupon_data');
var MINIMUM_SPEND_SELECTOR = _seleniumWebdriver.By.css('#minimum_amount');
var MAXIMUM_SPEND_SELECTOR = _seleniumWebdriver.By.css('#maximum_amount');
var INDIVIDUAL_USE_SELECTOR = _seleniumWebdriver.By.css('#individual_use');
var EXCLUDE_SALE_ITEMS_SELECTOR = _seleniumWebdriver.By.css('#exclude_sale_items');
var EMAIL_RESTRICTIONS_SELECTOR = _seleniumWebdriver.By.css('#customer_email');

var ComponentCouponDataPanelUsageRestriction = function (_Component) {
	_inherits(ComponentCouponDataPanelUsageRestriction, _Component);

	function ComponentCouponDataPanelUsageRestriction(driver) {
		_classCallCheck(this, ComponentCouponDataPanelUsageRestriction);

		return _possibleConstructorReturn(this, (ComponentCouponDataPanelUsageRestriction.__proto__ || Object.getPrototypeOf(ComponentCouponDataPanelUsageRestriction)).call(this, driver, PANEL_SELECTOR, { wait: false }));
	}

	_createClass(ComponentCouponDataPanelUsageRestriction, [{
		key: 'setMinimumSpend',
		value: function setMinimumSpend(minimum) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, MINIMUM_SPEND_SELECTOR, minimum);
		}
	}, {
		key: 'setMaximumSpend',
		value: function setMaximumSpend(maximum) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, MAXIMUM_SPEND_SELECTOR, maximum);
		}
	}, {
		key: 'checkIndividualUseOnly',
		value: function checkIndividualUseOnly() {
			return wcHelper.setCheckbox(this.driver, INDIVIDUAL_USE_SELECTOR);
		}
	}, {
		key: 'uncheckIndividualUseOnly',
		value: function uncheckIndividualUseOnly() {
			return wcHelper.unsetCheckbox(this.driver, INDIVIDUAL_USE_SELECTOR);
		}
	}, {
		key: 'checkExcludeSaleItems',
		value: function checkExcludeSaleItems() {
			return wcHelper.setCheckbox(this.driver, EXCLUDE_SALE_ITEMS_SELECTOR);
		}
	}, {
		key: 'uncheckExcludeSaleItems',
		value: function uncheckExcludeSaleItems() {
			return wcHelper.unsetCheckbox(this.driver, EXCLUDE_SALE_ITEMS_SELECTOR);
		}
	}, {
		key: 'setEmailRestrictions',
		value: function setEmailRestrictions(emails) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, EMAIL_RESTRICTIONS_SELECTOR, emails);
		}
	}]);

	return ComponentCouponDataPanelUsageRestriction;
}(_wpE2ePageObjects.Component);

exports.default = ComponentCouponDataPanelUsageRestriction;
module.exports = exports['default'];
