// wc-e2e-page-objects/lib/components/wp-admin/component-meta-box-order-data.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _helper = require('../../helper');

var wcHelper = _interopRequireWildcard(_helper);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal Dependencies
 */


var METABOX_SELECTOR = _seleniumWebdriver.By.css('#woocommerce-order-data');
var ORDER_DATE_SELECTOR = _seleniumWebdriver.By.name('order_date');
var ORDER_DATE_HOUR_SELECTOR = _seleniumWebdriver.By.name('order_date_hour');
var ORDER_DATE_MINUTE_SELECTOR = _seleniumWebdriver.By.name('order_date_minute');
var ORDER_STATUS_SELECTOR = _seleniumWebdriver.By.css('.wc-order-status .select2-selection__arrow');
var CUSTOMER_SELECTOR = _seleniumWebdriver.By.css('.wc-customer-user .select2-selection__arrow');

var ComponentMetaBoxOrderData = function (_ComponentMetaBox) {
	_inherits(ComponentMetaBoxOrderData, _ComponentMetaBox);

	function ComponentMetaBoxOrderData(driver) {
		_classCallCheck(this, ComponentMetaBoxOrderData);

		return _possibleConstructorReturn(this, (ComponentMetaBoxOrderData.__proto__ || Object.getPrototypeOf(ComponentMetaBoxOrderData)).call(this, driver, METABOX_SELECTOR));
	}

	_createClass(ComponentMetaBoxOrderData, [{
		key: 'selectOrderStatus',
		value: function selectOrderStatus(status) {
			return wcHelper.select2Option(this.driver, ORDER_STATUS_SELECTOR, status);
		}
	}, {
		key: 'hasOrderStatus',
		value: function hasOrderStatus(status) {
			var selector = _seleniumWebdriver.By.xpath('//span[@id="select2-order_status-container" and contains(., "' + status + '")]');
			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'selectCustomer',
		value: function selectCustomer(keyword, exactOption) {
			return wcHelper.select2OptionWithSearch(this.driver, CUSTOMER_SELECTOR, keyword, exactOption);
		}
	}, {
		key: 'setOrderDate',
		value: function setOrderDate(date) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ORDER_DATE_SELECTOR, date);
		}
	}, {
		key: 'setOrderDateHour',
		value: function setOrderDateHour(hour) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ORDER_DATE_HOUR_SELECTOR, hour);
		}
	}, {
		key: 'setOrderDateMinute',
		value: function setOrderDateMinute(minute) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, ORDER_DATE_MINUTE_SELECTOR, minute);
		}
	}]);

	return ComponentMetaBoxOrderData;
}(_wpE2ePageObjects.ComponentMetaBox);

exports.default = ComponentMetaBoxOrderData;
module.exports = exports['default'];
