// wc-e2e-page-objects/lib/components/wp-admin/component-product-data-attribute-row.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var ComponentProductDataAttributeRow = function (_Component) {
	_inherits(ComponentProductDataAttributeRow, _Component);

	function ComponentProductDataAttributeRow(driver, selector) {
		_classCallCheck(this, ComponentProductDataAttributeRow);

		return _possibleConstructorReturn(this, (ComponentProductDataAttributeRow.__proto__ || Object.getPrototypeOf(ComponentProductDataAttributeRow)).call(this, driver, selector));
	}

	_createClass(ComponentProductDataAttributeRow, [{
		key: 'setName',
		value: function setName(name) {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' input[name^="attribute_name"]');
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, selector, name);
		}
	}, {
		key: 'checkVisibleOnTheProductPage',
		value: function checkVisibleOnTheProductPage() {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' input[name^="attribute_visibility"]');
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, selector);
		}
	}, {
		key: 'uncheckVisibleOnTheProductPage',
		value: function uncheckVisibleOnTheProductPage() {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' input[name^="attribute_visibility"]');
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, selector);
		}
	}, {
		key: 'checkUsedForVariations',
		value: function checkUsedForVariations() {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' input[name^="attribute_variation"]');
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, selector);
		}
	}, {
		key: 'uncheckUsedForVariations',
		value: function uncheckUsedForVariations() {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' input[name^="attribute_variation"]');
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, selector);
		}
	}, {
		key: 'setValue',
		value: function setValue(value) {
			var selector = _seleniumWebdriver.By.css(this.selector.value + ' textarea[name^="attribute_values"]');
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, selector, value);
		}
	}, {
		key: 'toggle',
		value: function toggle() {
			this.driver.actions().mouseMove(this.driver.findElement(_seleniumWebdriver.By.css(this.selector.value + ' > h3'))).click().perform();
		}
	}, {
		key: 'remove',
		value: function remove() {
			this.driver.actions().mouseMove(this.driver.findElement(_seleniumWebdriver.By.css(this.selector.value + ' > h3 > .remove_row'))).click().perform();

			this.driver.switchTo().alert().accept();
		}
	}]);

	return ComponentProductDataAttributeRow;
}(_wpE2ePageObjects.Component);

exports.default = ComponentProductDataAttributeRow;
module.exports = exports['default'];
