// wc-e2e-page-objects/lib/components/wp-admin/component-product-data-panel-variations.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _helper = require('../../helper');

var wcHelper = _interopRequireWildcard(_helper);

var _componentProductDataVariationRow = require('./component-product-data-variation-row');

var _componentProductDataVariationRow2 = _interopRequireDefault(_componentProductDataVariationRow);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var PANEL_SELECTOR = _seleniumWebdriver.By.css('#product_attributes');
var ACTION_SELECTOR = _seleniumWebdriver.By.css('#field_to_edit.variation_actions');
var GO_SELECTOR = _seleniumWebdriver.By.css('.do_variation_action');
var SAVE_CHANGES_SELECTOR = _seleniumWebdriver.By.css('button.save-variation-changes');

var ComponentProductDataPanelVariations = function (_Component) {
	_inherits(ComponentProductDataPanelVariations, _Component);

	function ComponentProductDataPanelVariations(driver) {
		_classCallCheck(this, ComponentProductDataPanelVariations);

		var _this = _possibleConstructorReturn(this, (ComponentProductDataPanelVariations.__proto__ || Object.getPrototypeOf(ComponentProductDataPanelVariations)).call(this, driver, PANEL_SELECTOR, { wait: false }));

		_this.currentAction = 'Add variation';
		return _this;
	}

	_createClass(ComponentProductDataPanelVariations, [{
		key: 'selectAction',
		value: function selectAction(action) {
			wcHelper.waitTillUIBlockNotPresent(this.driver);

			this.currentAction = action;
			_wpE2eWebdriver.WebDriverHelper.selectOption(this.driver, ACTION_SELECTOR, action);
		}
	}, {
		key: 'go',
		value: function go() {
			wcHelper.waitTillUIBlockNotPresent(this.driver);

			_wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, GO_SELECTOR);

			switch (this.currentAction) {
				case 'Create variations from all attributes':
					wcHelper.waitTillAlertAccepted(this.driver);
					wcHelper.waitTillAlertAccepted(this.driver);
					break;
			}

			wcHelper.waitTillUIBlockNotPresent(this.driver);
		}
	}, {
		key: 'getRow',
		value: function getRow(row) {
			return new _componentProductDataVariationRow2.default(this.driver, this._getVariationRowSelector(row));
		}
	}, {
		key: 'saveChanges',
		value: function saveChanges() {
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, SAVE_CHANGES_SELECTOR);
		}
	}, {
		key: '_getVariationRowSelector',
		value: function _getVariationRowSelector(row) {
			var selector = void 0;
			switch (row) {
				case 'first':
					selector = _seleniumWebdriver.By.css('.woocommerce_variations .woocommerce_variation:first-child');
					break;
				case 'last':
					selector = _seleniumWebdriver.By.css('.woocommerce_variations .woocommerce_variation:last-child');
					break;
				default:
					row = parseInt(row, 10);
					selector = _seleniumWebdriver.By.css('.woocommerce_variations .woocommerce_variation:nth-child(' + row + ')');
			}

			return selector;
		}
	}]);

	return ComponentProductDataPanelVariations;
}(_wpE2ePageObjects.Component);

exports.default = ComponentProductDataPanelVariations;
module.exports = exports['default'];
