// wc-e2e-page-objects/lib/pages/shop-page.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _componentProductCard = require('../components/component-product-card.js');

var _componentProductCard2 = _interopRequireDefault(_componentProductCard);

var _componentProductsSorter = require('../components/component-products-sorter');

var _componentProductsSorter2 = _interopRequireDefault(_componentProductsSorter);

var _componentProductsPagination = require('../components/component-products-pagination');

var _componentProductsPagination2 = _interopRequireDefault(_componentProductsPagination);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * @module ShopPage
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

/**
 * External dependencies
 */


/**
 * Internal dependencies
 */


var defaultArgs = {
	visit: true,
	components: {
		sorter: _componentProductsSorter2.default,
		pagination: _componentProductsPagination2.default
	}
};

/**
 * The front-end Shop page.
 *
 * @extends Page
 */

var ShopPage = function (_Page) {
	_inherits(ShopPage, _Page);

	/**
 	* @param {WebDriver} driver   - Instance of WebDriver.
 	* @param {object}    args     - Configuration arguments.
 */
	function ShopPage(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, ShopPage);

		args = Object.assign(defaultArgs, args);
		return _possibleConstructorReturn(this, (ShopPage.__proto__ || Object.getPrototypeOf(ShopPage)).call(this, driver, args));
	}

	/**
 * Click a product.
 *
 	* @param  {string}    productTitle - The title of the product to click.
 	* @return {Promise}   Promise that evaluates to `true` if product is found and clicked, `false` otherwise.
 */


	_createClass(ShopPage, [{
		key: 'clickProduct',
		value: function clickProduct(productTitle) {
			var card = new _componentProductCard2.default(this.driver, productTitle);
			return card.click();
		}

		/**
  * Add a product to the cart.
  *
  	* @param  {string}    productTitle - The title of the product to add.
  	* @return {Promise}   Promise that evaluates to `true` if product is found and added, `false` otherwise.
  */

	}, {
		key: 'addProductToCart',
		value: function addProductToCart(productTitle) {
			var card = new _componentProductCard2.default(this.driver, productTitle);
			return card.addToCart();
		}

		/**
  * Change the shop product sorting
  *
  	* @param  {string}    optionText - The text of the sort option. Default: "Default sorting".
  	* @return {Promise}   Promise that evaluates to `true` if able to select sort option, `false` otherwise.
  */

	}, {
		key: 'sortBy',
		value: function sortBy() {
			var optionText = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'Default sorting';

			return this.components.sorter.sortBy(optionText);
		}

		/**
  * Go to a specific shop page
  *
  	* @param  {number}    pageNum - The page number to go to.
  	* @return {Promise}   Promise that evaluates to `true` if able to go to specified page, `false` otherwise.
  */

	}, {
		key: 'goToPage',
		value: function goToPage(pageNum) {
			return this.components.pagination.goToPage(pageNum);
		}

		/**
  * Go to the next shop page
  *
  	* @return {Promise}   Promise that evaluates to `true` if able to go to next page, `false` otherwise.
  */

	}, {
		key: 'nextPage',
		value: function nextPage() {
			return this.components.pagination.nextPage();
		}
	}]);

	return ShopPage;
}(_wpE2ePageObjects.Page);

exports.default = ShopPage;
module.exports = exports['default'];
