// wp-e2e-page-objects/lib/components/wp-admin/component-post-term-form-edit.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _component = require('../component');

var _component2 = _interopRequireDefault(_component);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * External dependencies
 */


var FORM_SELECTOR = _seleniumWebdriver.By.css('form[action="edit-tags.php"]');
var NAME_SELECTOR = _seleniumWebdriver.By.css('#name');
var SLUG_SELECTOR = _seleniumWebdriver.By.css('#slug');
var PARENT_SELECTOR = _seleniumWebdriver.By.css('#parent');
var DESCRIPTION_SELECTOR = _seleniumWebdriver.By.css('#description');
var SUBMIT_SELECTOR = _seleniumWebdriver.By.css('#submit');

var ComponentPostTermFormEdit = function (_Component) {
	_inherits(ComponentPostTermFormEdit, _Component);

	function ComponentPostTermFormEdit(driver) {
		var selector = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : FORM_SELECTOR;

		_classCallCheck(this, ComponentPostTermFormEdit);

		return _possibleConstructorReturn(this, (ComponentPostTermFormEdit.__proto__ || Object.getPrototypeOf(ComponentPostTermFormEdit)).call(this, driver, selector));
	}

	_createClass(ComponentPostTermFormEdit, [{
		key: 'setName',
		value: function setName(name) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, NAME_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, NAME_SELECTOR, name);
		}
	}, {
		key: 'setSlug',
		value: function setSlug(slug) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, SLUG_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, SLUG_SELECTOR, slug);
		}
	}, {
		key: 'selectParent',
		value: function selectParent(option) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, PARENT_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.selectOption(this.driver, PARENT_SELECTOR, option);
		}
	}, {
		key: 'setDescription',
		value: function setDescription(description) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, DESCRIPTION_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, DESCRIPTION_SELECTOR, description);
		}
	}, {
		key: 'submit',
		value: function submit() {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, SUBMIT_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, SUBMIT_SELECTOR);
		}
	}]);

	return ComponentPostTermFormEdit;
}(_component2.default);

exports.default = ComponentPostTermFormEdit;
module.exports = exports['default'];
