// wp-e2e-page-objects/lib/components/wp-admin/component-meta-box-publish.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _componentMetaBox = require('./component-meta-box');

var _componentMetaBox2 = _interopRequireDefault(_componentMetaBox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var METABOX_SELECTOR = _seleniumWebdriver.By.css('#submitdiv');
var SAVE_SELECTOR = _seleniumWebdriver.By.css('#save-post');
var PUBLISH_SELECTOR = _seleniumWebdriver.By.css('#publish');
var EDIT_STATUS_SELECTOR = _seleniumWebdriver.By.css('.edit-post-status');
var STATUS_SELECTOR = _seleniumWebdriver.By.css('#post_status');
var SAVE_STATUS_SELECTOR = _seleniumWebdriver.By.css('.save-post-status');
var MOVE_TO_TRASH_SELECTOR = _seleniumWebdriver.By.css('.submitdelete');

var ComponentMetaBoxPublish = function (_ComponentMetaBox) {
	_inherits(ComponentMetaBoxPublish, _ComponentMetaBox);

	function ComponentMetaBoxPublish(driver) {
		_classCallCheck(this, ComponentMetaBoxPublish);

		return _possibleConstructorReturn(this, (ComponentMetaBoxPublish.__proto__ || Object.getPrototypeOf(ComponentMetaBoxPublish)).call(this, driver, METABOX_SELECTOR, { wait: false }));
	}

	_createClass(ComponentMetaBoxPublish, [{
		key: 'publish',
		value: function publish() {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, PUBLISH_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, PUBLISH_SELECTOR);
		}
	}, {
		key: 'save',
		value: function save() {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, SAVE_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, SAVE_SELECTOR);
		}
	}, {
		key: 'selectStatus',
		value: function selectStatus(status) {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, EDIT_STATUS_SELECTOR);
			_wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, EDIT_STATUS_SELECTOR);
			_wpE2eWebdriver.WebDriverHelper.selectOption(this.driver, STATUS_SELECTOR, status);
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, SAVE_STATUS_SELECTOR);
		}
	}, {
		key: 'hasStatus',
		value: function hasStatus(status) {
			var selector = _seleniumWebdriver.By.xpath('//span[@id="post-status-display" and contains(text(), "' + status + '")]');
			return _wpE2eWebdriver.WebDriverHelper.isEventuallyPresentAndDisplayed(this.driver, selector);
		}
	}, {
		key: 'moveToTrash',
		value: function moveToTrash() {
			_wpE2eWebdriver.WebDriverHelper.mouseMoveTo(this.driver, MOVE_TO_TRASH_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, MOVE_TO_TRASH_SELECTOR);
		}
	}]);

	return ComponentMetaBoxPublish;
}(_componentMetaBox2.default);

exports.default = ComponentMetaBoxPublish;
module.exports = exports['default'];
