// wp-e2e-page-objects/lib/components/wp-admin/component-login-form.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _deprecate = require('deprecate');

var _deprecate2 = _interopRequireDefault(_deprecate);

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _component = require('../component');

var _component2 = _interopRequireDefault(_component);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * External dependencies
 */


var LOGIN_FORM_SELECTOR = _seleniumWebdriver.By.css('#loginform');
var USERNAME_FIELD_SELECTOR = _seleniumWebdriver.By.css('#user_login');
var PASSWORD_FIELD_SELECTOR = _seleniumWebdriver.By.css('#user_pass');
var LOGIN_BUTTON_SELECTOR = _seleniumWebdriver.By.css('#wp-submit');

/**
 * Login form component.
 *
 * @extends Component
 */

var ComponentLoginForm = function (_Component) {
	_inherits(ComponentLoginForm, _Component);

	function ComponentLoginForm(driver) {
		var selector = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : LOGIN_FORM_SELECTOR;

		_classCallCheck(this, ComponentLoginForm);

		return _possibleConstructorReturn(this, (ComponentLoginForm.__proto__ || Object.getPrototypeOf(ComponentLoginForm)).call(this, driver, selector));
	}

	/**
  * Set username field.
  *
  * @param {string} username - Username.
  *
  * @return {Promise} Promise that evaluates to `true` if username field is
  *                   set successfully, `false` otherwise.
  */


	_createClass(ComponentLoginForm, [{
		key: 'setUsername',
		value: function setUsername(username) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, USERNAME_FIELD_SELECTOR, username);
		}

		/**
   * Set password field.
   *
   * @param {string} password - Password.
   *
   * @return {Promise} Promise that evaluates to `true` if password field is
   *                   set successfully, `false` otherwise.
   */

	}, {
		key: 'setPassword',
		value: function setPassword(password) {
			return _wpE2eWebdriver.WebDriverHelper.setWhenSettable(this.driver, PASSWORD_FIELD_SELECTOR, password);
		}

		/**
   * Login.
   *
   * @return {Promise} Promise that evaluates to `true` if login button is
   *                   clicked successfully.
   */

	}, {
		key: 'login',
		value: function login() {
			return _wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, LOGIN_BUTTON_SELECTOR);
		}

		/**
   * Set username field.
   *
   * @deprecated Since 0.7.0
   *
   * @param {string} username - Username.
   *
   * @return {Promise} Promise that evaluates to `true` if username field is
   *                   set successfully, `false` otherwise.
   */

	}, {
		key: 'fillUsername',
		value: function fillUsername(username) {
			(0, _deprecate2.default)('ComponentLoginForm.fillUsername() is deprecated, use ComponentLoginForm.setUsername() instead');
			return this.setUsername(username);
		}

		/**
   * Set password field.
   *
   * @deprecated Since 0.7.0
   *
   * @param {string} password - Password.
   *
   * @return {Promise} Promise that evaluates to `true` if password field is
   *                   set successfully, `false` otherwise.
   */

	}, {
		key: 'fillPassword',
		value: function fillPassword(password) {
			(0, _deprecate2.default)('ComponentLoginForm.fillPassword() is deprecated, use ComponentLoginForm.setPassword() instead');
			return this.setPassword(password);
		}

		/**
   * Login.
   *
   * @deprecated Since 0.7.0
   *
   * @return {Promise} Promise that evaluates to `true` if login button is
   *                   clicked successfully.
   */

	}, {
		key: 'submit',
		value: function submit() {
			(0, _deprecate2.default)('ComponentLoginForm.submit() is deprecated, use ComponentLoginForm.login instead');
			return this.login();
		}
	}]);

	return ComponentLoginForm;
}(_component2.default);

exports.default = ComponentLoginForm;
module.exports = exports['default'];
