// wp-e2e-page-objects/lib/flows/user-flow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */


/**
 * Internal dependencies
 */


var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _pageMap = require('../page-map');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var UserFlow = function () {
	function UserFlow(driver, _ref) {
		var _ref$baseUrl = _ref.baseUrl,
		    baseUrl = _ref$baseUrl === undefined ? 'http://example.com' : _ref$baseUrl,
		    _ref$username = _ref.username,
		    username = _ref$username === undefined ? '' : _ref$username,
		    _ref$password = _ref.password,
		    password = _ref$password === undefined ? '' : _ref$password;

		_classCallCheck(this, UserFlow);

		this.driver = driver;
		this.baseUrl = baseUrl;
		this.username = username;
		this.password = password;

		_wpE2eWebdriver.WebDriverHelper.clearCookiesAndDeleteLocalStorage(this.driver);

		var loginPage = this.open(_pageMap.PAGE.WP_LOGIN);
		this.currentPage = loginPage.login(this.username, this.password);
	}

	_createClass(UserFlow, [{
		key: 'open',
		value: function open(page) {
			if ('object' !== (typeof page === 'undefined' ? 'undefined' : _typeof(page))) {
				throw new Error('page should be an object containing `path` and `oject`.');
			}

			var PageObject = page.object;

			for (var _len = arguments.length, args = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
				args[_key - 1] = arguments[_key];
			}

			var pageArgs = {
				url: _pageMap.getPageUrl.apply(undefined, [this.baseUrl, page].concat(args)),
				visit: true
			};

			this.currentPage = new PageObject(this.driver, pageArgs);

			return this.currentPage;
		}
	}, {
		key: 'createPost',
		value: function createPost(post) {
			post = Object.assign({
				title: '',
				status: 'Published'
			}, post);

			var newPostPage = this.open(_pageMap.PAGE.WP_ADMIN_NEW_POST);
			this._setPost(newPostPage, post);

			return post.status === 'Published' ? newPostPage.publish() : newPostPage.save();
		}
	}, {
		key: '_setPost',
		value: function _setPost(page, post) {
			// TODO(gedex): set content, status, etc.

			if (post.title) {
				page.setTitle(post.title);
			}

			if (post.status) {
				page.selectStatus(post.status);
			}
		}
	}, {
		key: 'createCategory',
		value: function createCategory(category) {
			category = Object.assign({ name: '' }, category);

			var categoriesPage = this.open(_pageMap.PAGE.WP_ADMIN_CATEGORIES);
			this._setCategory(categoriesPage, category);

			return categoriesPage.add();
		}
	}, {
		key: '_setCategory',
		value: function _setCategory(page, category) {
			if (category.name) {
				page.setName(category.name);
			}

			if (category.slug) {
				page.setSlug(category.slug);
			}

			if (category.parent) {
				page.selectParent(category.parent);
			}

			if (category.description) {
				page.setDescription(category.description);
			}
		}
	}, {
		key: 'createTag',
		value: function createTag(tag) {
			tag = Object.assign({ name: '' }, tag);

			var tagsPage = this.open(_pageMap.PAGE.WP_ADMIN_TAGS);
			this._setTag(tagsPage, tag);

			return tagsPage.add();
		}
	}, {
		key: '_setTag',
		value: function _setTag(page, tag) {
			if (tag.name) {
				page.setName(tag.name);
			}

			if (tag.slug) {
				page.setSlug(tag.slug);
			}

			if (tag.description) {
				page.setDescription(tag.description);
			}
		}
	}, {
		key: 'addComment',
		value: function addComment(postTitle, comment) {
			var postsList = this.open(_pageMap.PAGE.WP_ADMIN_POSTS);
			var post = postsList.viewPostWithTitle(postTitle);

			return post.postComment(comment);
		}
	}, {
		key: 'logout',
		value: function logout() {
			return this.currentPage.hoverMyAccountThenClickSubmenu('Log Out');
		}
	}]);

	return UserFlow;
}();

exports.default = UserFlow;
module.exports = exports['default'];
