// wp-e2e-page-objects/lib/flows/visitor-flow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */


/**
 * Internal dependencies
 */


var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _pageMap = require('../page-map');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var UserFlow = function () {
	function UserFlow(driver, _ref) {
		var _ref$baseUrl = _ref.baseUrl,
		    baseUrl = _ref$baseUrl === undefined ? 'http://example.com' : _ref$baseUrl;

		_classCallCheck(this, UserFlow);

		this.driver = driver;
		this.baseUrl = baseUrl;

		_wpE2eWebdriver.WebDriverHelper.clearCookiesAndDeleteLocalStorage(this.driver);
	}

	_createClass(UserFlow, [{
		key: 'open',
		value: function open(page) {
			if ('object' !== (typeof page === 'undefined' ? 'undefined' : _typeof(page))) {
				throw new Error('page should be an object containing `path` and `oject`.');
			}

			var PageObject = page.object;

			for (var _len = arguments.length, args = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
				args[_key - 1] = arguments[_key];
			}

			var pageArgs = {
				url: _pageMap.getPageUrl.apply(undefined, [this.baseUrl, page].concat(args)),
				visit: true
			};

			this.currentPage = new PageObject(this.driver, pageArgs);

			return this.currentPage;
		}
	}]);

	return UserFlow;
}();

exports.default = UserFlow;
module.exports = exports['default'];
