// wp-e2e-page-objects/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.VisitorFlow = exports.UserFlow = exports.WPLogin = exports.WPAdminTagEdit = exports.WPAdminTags = exports.WPAdminSettingsGeneral = exports.WPAdminSettings = exports.WPAdminPostTermEdit = exports.WPAdminPostTerms = exports.WPAdminPosts = exports.WPAdminPostNew = exports.WPAdminPostEdit = exports.WPAdminPlugins = exports.WPAdminPluginInstall = exports.WPAdminDashboard = exports.WPAdminCategoryEdit = exports.WPAdminCategories = exports.WPAdmin = exports.SinglePage = exports.FrontPage = exports.CategoryPage = exports.ArchivePage = exports.Page = exports.PageMap = exports.ComponentWelcomePanel = exports.ComponentPostTermsList = exports.ComponentPostTermFormNew = exports.ComponentPostTermFormEdit = exports.ComponentPluginsList = exports.ComponentPluginInstallFilter = exports.ComponentPluginCard = exports.ComponentMetaBoxPublish = exports.ComponentMetaBoxFormat = exports.ComponentMetaBoxCategories = exports.ComponentMetaBox = exports.ComponentAdminNotice = exports.ComponentAdminMenu = exports.ComponentAdminBarQuickLinks = exports.ComponentAdminBarMyAccount = exports.ComponentAdminBar = exports.ComponentLoginForm = exports.ComponentWidgetRecentComments = exports.ComponentWidgetRecentPosts = exports.ComponentWidget = exports.ComponentSidebar = exports.ComponentComment = exports.Component = undefined;

var _component = require('./components/component');

var _component2 = _interopRequireDefault(_component);

var _componentComment = require('./components/component-comment');

var _componentComment2 = _interopRequireDefault(_componentComment);

var _componentSidebar = require('./components/component-sidebar');

var _componentSidebar2 = _interopRequireDefault(_componentSidebar);

var _componentWidget = require('./components/component-widget');

var _componentWidget2 = _interopRequireDefault(_componentWidget);

var _componentWidgetRecentPosts = require('./components/component-widget-recent-posts');

var _componentWidgetRecentPosts2 = _interopRequireDefault(_componentWidgetRecentPosts);

var _componentWidgetRecentComments = require('./components/component-widget-recent-comments');

var _componentWidgetRecentComments2 = _interopRequireDefault(_componentWidgetRecentComments);

var _componentLoginForm = require('./components/wp-admin/component-login-form');

var _componentLoginForm2 = _interopRequireDefault(_componentLoginForm);

var _componentAdminBar = require('./components/wp-admin/component-admin-bar');

var _componentAdminBar2 = _interopRequireDefault(_componentAdminBar);

var _componentAdminBarMyAccount = require('./components/wp-admin/component-admin-bar-my-account');

var _componentAdminBarMyAccount2 = _interopRequireDefault(_componentAdminBarMyAccount);

var _componentAdminBarQuickLinks = require('./components/wp-admin/component-admin-bar-quick-links');

var _componentAdminBarQuickLinks2 = _interopRequireDefault(_componentAdminBarQuickLinks);

var _componentAdminMenu = require('./components/wp-admin/component-admin-menu');

var _componentAdminMenu2 = _interopRequireDefault(_componentAdminMenu);

var _componentAdminNotice = require('./components/wp-admin/component-admin-notice');

var _componentAdminNotice2 = _interopRequireDefault(_componentAdminNotice);

var _componentMetaBox = require('./components/wp-admin/component-meta-box');

var _componentMetaBox2 = _interopRequireDefault(_componentMetaBox);

var _componentMetaBoxCategories = require('./components/wp-admin/component-meta-box-categories');

var _componentMetaBoxCategories2 = _interopRequireDefault(_componentMetaBoxCategories);

var _componentMetaBoxFormat = require('./components/wp-admin/component-meta-box-format');

var _componentMetaBoxFormat2 = _interopRequireDefault(_componentMetaBoxFormat);

var _componentMetaBoxPublish = require('./components/wp-admin/component-meta-box-publish');

var _componentMetaBoxPublish2 = _interopRequireDefault(_componentMetaBoxPublish);

var _componentPluginCard = require('./components/wp-admin/component-plugin-card');

var _componentPluginCard2 = _interopRequireDefault(_componentPluginCard);

var _componentPluginInstallFilter = require('./components/wp-admin/component-plugin-install-filter');

var _componentPluginInstallFilter2 = _interopRequireDefault(_componentPluginInstallFilter);

var _componentPluginsList = require('./components/wp-admin/component-plugins-list');

var _componentPluginsList2 = _interopRequireDefault(_componentPluginsList);

var _componentPostTermFormEdit = require('./components/wp-admin/component-post-term-form-edit');

var _componentPostTermFormEdit2 = _interopRequireDefault(_componentPostTermFormEdit);

var _componentPostTermFormNew = require('./components/wp-admin/component-post-term-form-new');

var _componentPostTermFormNew2 = _interopRequireDefault(_componentPostTermFormNew);

var _componentPostTermsList = require('./components/wp-admin/component-post-terms-list');

var _componentPostTermsList2 = _interopRequireDefault(_componentPostTermsList);

var _componentWelcomePanel = require('./components/wp-admin/component-welcome-panel');

var _componentWelcomePanel2 = _interopRequireDefault(_componentWelcomePanel);

var _pageMap = require('./page-map');

var PageMap = _interopRequireWildcard(_pageMap);

var _page = require('./pages/page');

var _page2 = _interopRequireDefault(_page);

var _archivePage = require('./pages/archive-page');

var _archivePage2 = _interopRequireDefault(_archivePage);

var _categoryPage = require('./pages/category-page');

var _categoryPage2 = _interopRequireDefault(_categoryPage);

var _frontPage = require('./pages/front-page');

var _frontPage2 = _interopRequireDefault(_frontPage);

var _singlePage = require('./pages/single-page');

var _singlePage2 = _interopRequireDefault(_singlePage);

var _wpAdmin = require('./pages/wp-admin/wp-admin');

var _wpAdmin2 = _interopRequireDefault(_wpAdmin);

var _wpAdminCategories = require('./pages/wp-admin/wp-admin-categories');

var _wpAdminCategories2 = _interopRequireDefault(_wpAdminCategories);

var _wpAdminCategoryEdit = require('./pages/wp-admin/wp-admin-category-edit');

var _wpAdminCategoryEdit2 = _interopRequireDefault(_wpAdminCategoryEdit);

var _wpAdminDashboard = require('./pages/wp-admin/wp-admin-dashboard');

var _wpAdminDashboard2 = _interopRequireDefault(_wpAdminDashboard);

var _wpAdminPluginInstall = require('./pages/wp-admin/wp-admin-plugin-install');

var _wpAdminPluginInstall2 = _interopRequireDefault(_wpAdminPluginInstall);

var _wpAdminPlugins = require('./pages/wp-admin/wp-admin-plugins');

var _wpAdminPlugins2 = _interopRequireDefault(_wpAdminPlugins);

var _wpAdminPostEdit = require('./pages/wp-admin/wp-admin-post-edit');

var _wpAdminPostEdit2 = _interopRequireDefault(_wpAdminPostEdit);

var _wpAdminPostNew = require('./pages/wp-admin/wp-admin-post-new');

var _wpAdminPostNew2 = _interopRequireDefault(_wpAdminPostNew);

var _wpAdminPosts = require('./pages/wp-admin/wp-admin-posts');

var _wpAdminPosts2 = _interopRequireDefault(_wpAdminPosts);

var _wpAdminPostTerms = require('./pages/wp-admin/wp-admin-post-terms');

var _wpAdminPostTerms2 = _interopRequireDefault(_wpAdminPostTerms);

var _wpAdminPostTermEdit = require('./pages/wp-admin/wp-admin-post-term-edit');

var _wpAdminPostTermEdit2 = _interopRequireDefault(_wpAdminPostTermEdit);

var _wpAdminSettings = require('./pages/wp-admin/wp-admin-settings');

var _wpAdminSettings2 = _interopRequireDefault(_wpAdminSettings);

var _wpAdminSettingsGeneral = require('./pages/wp-admin/wp-admin-settings-general');

var _wpAdminSettingsGeneral2 = _interopRequireDefault(_wpAdminSettingsGeneral);

var _wpAdminTags = require('./pages/wp-admin/wp-admin-tags');

var _wpAdminTags2 = _interopRequireDefault(_wpAdminTags);

var _wpAdminTagEdit = require('./pages/wp-admin/wp-admin-tag-edit');

var _wpAdminTagEdit2 = _interopRequireDefault(_wpAdminTagEdit);

var _wpLogin = require('./pages/wp-admin/wp-login');

var _wpLogin2 = _interopRequireDefault(_wpLogin);

var _userFlow = require('./flows/user-flow');

var _userFlow2 = _interopRequireDefault(_userFlow);

var _visitorFlow = require('./flows/visitor-flow');

var _visitorFlow2 = _interopRequireDefault(_visitorFlow);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Component = _component2.default;
exports.ComponentComment = _componentComment2.default;
exports.ComponentSidebar = _componentSidebar2.default;
exports.ComponentWidget = _componentWidget2.default;
exports.ComponentWidgetRecentPosts = _componentWidgetRecentPosts2.default;
exports.ComponentWidgetRecentComments = _componentWidgetRecentComments2.default;
exports.ComponentLoginForm = _componentLoginForm2.default;
exports.ComponentAdminBar = _componentAdminBar2.default;
exports.ComponentAdminBarMyAccount = _componentAdminBarMyAccount2.default;
exports.ComponentAdminBarQuickLinks = _componentAdminBarQuickLinks2.default;
exports.ComponentAdminMenu = _componentAdminMenu2.default;
exports.ComponentAdminNotice = _componentAdminNotice2.default;
exports.ComponentMetaBox = _componentMetaBox2.default;
exports.ComponentMetaBoxCategories = _componentMetaBoxCategories2.default;
exports.ComponentMetaBoxFormat = _componentMetaBoxFormat2.default;
exports.ComponentMetaBoxPublish = _componentMetaBoxPublish2.default;
exports.ComponentPluginCard = _componentPluginCard2.default;
exports.ComponentPluginInstallFilter = _componentPluginInstallFilter2.default;
exports.ComponentPluginsList = _componentPluginsList2.default;
exports.ComponentPostTermFormEdit = _componentPostTermFormEdit2.default;
exports.ComponentPostTermFormNew = _componentPostTermFormNew2.default;
exports.ComponentPostTermsList = _componentPostTermsList2.default;
exports.ComponentWelcomePanel = _componentWelcomePanel2.default;
exports.PageMap = PageMap;
exports.Page = _page2.default;
exports.ArchivePage = _archivePage2.default;
exports.CategoryPage = _categoryPage2.default;
exports.FrontPage = _frontPage2.default;
exports.SinglePage = _singlePage2.default;
exports.WPAdmin = _wpAdmin2.default;
exports.WPAdminCategories = _wpAdminCategories2.default;
exports.WPAdminCategoryEdit = _wpAdminCategoryEdit2.default;
exports.WPAdminDashboard = _wpAdminDashboard2.default;
exports.WPAdminPluginInstall = _wpAdminPluginInstall2.default;
exports.WPAdminPlugins = _wpAdminPlugins2.default;
exports.WPAdminPostEdit = _wpAdminPostEdit2.default;
exports.WPAdminPostNew = _wpAdminPostNew2.default;
exports.WPAdminPosts = _wpAdminPosts2.default;
exports.WPAdminPostTerms = _wpAdminPostTerms2.default;
exports.WPAdminPostTermEdit = _wpAdminPostTermEdit2.default;
exports.WPAdminSettings = _wpAdminSettings2.default;
exports.WPAdminSettingsGeneral = _wpAdminSettingsGeneral2.default;
exports.WPAdminTags = _wpAdminTags2.default;
exports.WPAdminTagEdit = _wpAdminTagEdit2.default;
exports.WPLogin = _wpLogin2.default;
exports.UserFlow = _userFlow2.default;
exports.VisitorFlow = _visitorFlow2.default; /**
                                              * Internal Dependencies
                                              */
