// wp-e2e-page-objects/lib/page-map.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.PAGE = undefined;
exports.getPageUrl = getPageUrl;

var _urlJoin = require('url-join');

var _urlJoin2 = _interopRequireDefault(_urlJoin);

var _sprintfJs = require('sprintf-js');

var _frontPage = require('./pages/front-page');

var _frontPage2 = _interopRequireDefault(_frontPage);

var _wpAdmin = require('./pages/wp-admin/wp-admin');

var _wpAdmin2 = _interopRequireDefault(_wpAdmin);

var _wpAdminDashboard = require('./pages/wp-admin/wp-admin-dashboard');

var _wpAdminDashboard2 = _interopRequireDefault(_wpAdminDashboard);

var _wpAdminCategories = require('./pages/wp-admin/wp-admin-categories');

var _wpAdminCategories2 = _interopRequireDefault(_wpAdminCategories);

var _wpAdminPluginInstall = require('./pages/wp-admin/wp-admin-plugin-install');

var _wpAdminPluginInstall2 = _interopRequireDefault(_wpAdminPluginInstall);

var _wpAdminPlugins = require('./pages/wp-admin/wp-admin-plugins');

var _wpAdminPlugins2 = _interopRequireDefault(_wpAdminPlugins);

var _wpAdminPostEdit = require('./pages/wp-admin/wp-admin-post-edit');

var _wpAdminPostEdit2 = _interopRequireDefault(_wpAdminPostEdit);

var _wpAdminPostNew = require('./pages/wp-admin/wp-admin-post-new');

var _wpAdminPostNew2 = _interopRequireDefault(_wpAdminPostNew);

var _wpAdminPosts = require('./pages/wp-admin/wp-admin-posts');

var _wpAdminPosts2 = _interopRequireDefault(_wpAdminPosts);

var _wpAdminSettingsGeneral = require('./pages/wp-admin/wp-admin-settings-general');

var _wpAdminSettingsGeneral2 = _interopRequireDefault(_wpAdminSettingsGeneral);

var _wpAdminTags = require('./pages/wp-admin/wp-admin-tags');

var _wpAdminTags2 = _interopRequireDefault(_wpAdminTags);

var _wpLogin = require('./pages/wp-admin/wp-login');

var _wpLogin2 = _interopRequireDefault(_wpLogin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Internal dependencies
 */
/**
 * External dependencies
 */
var PAGE = exports.PAGE = {
	FRONT_PAGE: {
		object: _frontPage2.default,
		path: '/'
	},
	WP_LOGIN: {
		object: _wpLogin2.default,
		path: '/wp-login.php'
	},
	WP_ADMIN: {
		object: _wpAdmin2.default,
		path: '/wp-admin/'
	},
	WP_ADMIN_DASHBOARD: {
		object: _wpAdminDashboard2.default,
		path: '/wp-admin/'
	},
	WP_ADMIN_POSTS: {
		object: _wpAdminPosts2.default,
		path: '/wp-admin/edit.php'
	},
	WP_ADMIN_NEW_POST: {
		object: _wpAdminPostNew2.default,
		path: '/wp-admin/post-new.php'
	},
	WP_ADMIN_EDIT_POST: {
		object: _wpAdminPostEdit2.default,
		path: '/wp-admin/post.php?post=%s&action=edit'
	},
	WP_ADMIN_CATEGORIES: {
		object: _wpAdminCategories2.default,
		path: '/wp-admin/edit-tags.php?taxonomy=category'
	},
	WP_ADMIN_TAGS: {
		object: _wpAdminTags2.default,
		path: '/wp-admin/edit-tags.php?taxonomy=post_tag'
	},
	WP_ADMIN_PAGES: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/edit.php?post_type=page'
	},
	WP_ADMIN_NEW_PAGE: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/post-new.php?post_type=page'
	},
	WP_ADMIN_EDIT_PAGE: {
		object: null,
		path: '/wp-admin/post.php?post=%s&action=edit'
	},
	WP_ADMIN_MEDIA: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/upload.php'
	},
	WP_ADMIN_NEW_MEDIA: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/media-new.php'
	},
	WP_ADMIN_COMMENTS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/edit-comments.php'
	},
	WP_ADMIN_EDIT_COMMENT: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/comment.php?action=editcomment&c=%s'
	},
	WP_ADMIN_THEMES: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/themes.php'
	},
	WP_ADMIN_NEW_THEME: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/theme-install.php'
	},
	WP_ADMIN_CUSTOMIZE: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/customize.php'
	},
	WP_ADMIN_WIDGETS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/widgets.php'
	},
	WP_ADMIN_EDIT_MENUS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/nav-menus.php'
	},
	WP_ADMIN_MANAGE_MENU_LOCATIONS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/nav-menus.php?action=locations'
	},
	WP_ADMIN_THEME_EDITOR: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/theme-editor.php'
	},
	WP_ADMIN_PLUGINS: {
		object: _wpAdminPlugins2.default,
		path: '/wp-admin/plugins.php'
	},
	WP_ADMIN_NEW_PLUGIN: {
		object: _wpAdminPluginInstall2.default,
		path: '/wp-admin/plugin-install.php'
	},
	WP_ADMIN_NEW_PLUGIN_UPLOAD: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/plugin-install.php?tab=upload'
	},
	WP_ADMIN_PLUGIN_EDITOR: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/plugin-editor.php'
	},
	WP_ADMIN_USERS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/users.php'
	},
	WP_ADMIN_NEW_USER: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/user-new.php'
	},
	WP_ADMIN_EDIT_USER: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/user-edit.php?user_id=%s'
	},
	WP_ADMIN_YOUR_PROFILE: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/profile.php'
	},
	WP_ADMIN_TOOLS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/tools.php'
	},
	WP_ADMIN_IMPORT: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/import.php'
	},
	WP_ADMIN_EXPORT: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/export.php'
	},
	WP_ADMIN_SETTINGS_GENERAL: {
		object: _wpAdminSettingsGeneral2.default,
		path: '/wp-admin/options-general.php'
	},
	WP_ADMIN_SETTINGS_WRITING: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-writing.php'
	},
	WP_ADMIN_SETTINGS_READING: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-reading.php'
	},
	WP_ADMIN_SETTINGS_DISCUSSION: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-discussion.php'
	},
	WP_ADMIN_SETTINGS_MEDIA: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-media.php'
	},
	WP_ADMIN_SETTINGS_PERMALINKS: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-permalink.php'
	},
	WP_ADMIN_SETTINGS_PERMALINK: {
		object: null, // TODO: Implement me!
		path: '/wp-admin/options-permalink.php'
	}
};

function getPageUrl(baseUrl, page) {
	for (var _len = arguments.length, args = Array(_len > 2 ? _len - 2 : 0), _key = 2; _key < _len; _key++) {
		args[_key - 2] = arguments[_key];
	}

	return (0, _urlJoin2.default)(baseUrl, _sprintfJs.sprintf.apply(undefined, [page.path].concat(args)));
}
