// wp-e2e-page-objects/lib/pages/wp-admin/wp-admin-posts.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _componentPostsList = require('../../components/wp-admin/component-posts-list');

var _componentPostsList2 = _interopRequireDefault(_componentPostsList);

var _singlePage = require('../single-page');

var _singlePage2 = _interopRequireDefault(_singlePage);

var _wpAdmin = require('./wp-admin');

var _wpAdmin2 = _interopRequireDefault(_wpAdmin);

var _wpAdminPostEdit = require('./wp-admin-post-edit');

var _wpAdminPostEdit2 = _interopRequireDefault(_wpAdminPostEdit);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Internal Dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var components = {
	postsList: _componentPostsList2.default
};

var defaultArgs = {
	url: '',
	visit: true,
	components: components
};

var WPAdminPosts = function (_WPAdmin) {
	_inherits(WPAdminPosts, _WPAdmin);

	function WPAdminPosts(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdminPosts);

		args = Object.assign(defaultArgs, args);
		return _possibleConstructorReturn(this, (WPAdminPosts.__proto__ || Object.getPrototypeOf(WPAdminPosts)).call(this, driver, args));
	}

	_createClass(WPAdminPosts, [{
		key: 'search',
		value: function search(keyword) {
			this.components.postsList.search(keyword);
			return this;
		}
	}, {
		key: 'editPostWithTitle',
		value: function editPostWithTitle(title) {
			this.components.postsList.editPostWithTitle(title);
			return new _wpAdminPostEdit2.default(this.driver, { visit: false });
		}
	}, {
		key: 'viewPostWithTitle',
		value: function viewPostWithTitle(title) {
			this.components.postsList.viewPostWithTitle(title);
			return new _singlePage2.default(this.driver, { visit: false });
		}
	}, {
		key: 'trashPostWithTitle',
		value: function trashPostWithTitle(title) {
			this.components.postsList.trashPostWithTitle(title);
			return this;
		}
	}]);

	return WPAdminPosts;
}(_wpAdmin2.default);

exports.default = WPAdminPosts;
module.exports = exports['default'];
