// wp-e2e-page-objects/lib/pages/wp-admin/wp-admin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _componentAdminMenu = require('../../components/wp-admin/component-admin-menu');

var _componentAdminMenu2 = _interopRequireDefault(_componentAdminMenu);

var _componentAdminBar = require('../../components/wp-admin/component-admin-bar');

var _componentAdminBar2 = _interopRequireDefault(_componentAdminBar);

var _componentAdminNotice = require('../../components/wp-admin/component-admin-notice');

var _componentAdminNotice2 = _interopRequireDefault(_componentAdminNotice);

var _page = require('../page');

var _page2 = _interopRequireDefault(_page);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Internal dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


function _getDefaultComponents() {
	return {
		adminMenu: _componentAdminMenu2.default,
		adminBar: _componentAdminBar2.default,
		adminNotice: _componentAdminNotice2.default
	};
}

var WPAdmin = function (_Page) {
	_inherits(WPAdmin, _Page);

	function WPAdmin(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdmin);

		args = Object.assign({ visit: false, url: '', components: {} }, args);
		args.components = Object.assign(_getDefaultComponents(), args.components);
		return _possibleConstructorReturn(this, (WPAdmin.__proto__ || Object.getPrototypeOf(WPAdmin)).call(this, driver, args));
	}

	_createClass(WPAdmin, [{
		key: 'hasMenu',
		value: function hasMenu(menu) {
			return this.components.adminMenu.hasMenu(menu);
		}
	}, {
		key: 'hasMenuWithSubmenu',
		value: function hasMenuWithSubmenu(menu, submenu) {
			return this.components.adminMenu.hasMenuWithSubmenu(menu, submenu);
		}
	}, {
		key: 'clickMenu',
		value: function clickMenu(menu) {
			return this.components.adminMenu.click(menu);
		}
	}, {
		key: 'hoverMenuThenClickSubmenu',
		value: function hoverMenuThenClickSubmenu(menu, submenu) {
			return this.components.adminMenu.hoverMenuThenClickSubmenu(menu, submenu);
		}
	}, {
		key: 'hasQuickLink',
		value: function hasQuickLink(menu) {
			return this.components.adminBar.quickLinks.hasMenu(menu);
		}
	}, {
		key: 'hasQuickLinkWithSubmenu',
		value: function hasQuickLinkWithSubmenu(menu, submenu) {
			return this.components.adminBar.quickLinks.hasMenuWithSubmenu(menu, submenu);
		}
	}, {
		key: 'clickQuickLink',
		value: function clickQuickLink(quickLink) {
			return this.components.adminBar.quickLinks.click(quickLink);
		}
	}, {
		key: 'hoverQuickLinkThenClickSubmenu',
		value: function hoverQuickLinkThenClickSubmenu(menu, submenu) {
			return this.components.adminBar.quickLinks.hoverMenuThenClickSubmenu(menu, submenu);
		}
	}, {
		key: 'hasMyAccountWithSubmenu',
		value: function hasMyAccountWithSubmenu(submenu) {
			return this.components.adminBar.myAccount.hasSubmenu(submenu);
		}
	}, {
		key: 'clickMyAccount',
		value: function clickMyAccount() {
			return this.components.adminBar.myAccount.click();
		}
	}, {
		key: 'hoverMyAccountThenClickSubmenu',
		value: function hoverMyAccountThenClickSubmenu(submenu) {
			return this.components.adminBar.myAccount.hoverMyAccountThenClickSubmenu(submenu);
		}
	}, {
		key: 'hasNotice',
		value: function hasNotice(notice) {
			return this.components.adminNotice.hasNotice(notice);
		}
	}, {
		key: 'isNoticeDismissible',
		value: function isNoticeDismissible(notice) {
			return this.components.adminNotice.isDismissible(notice);
		}
	}, {
		key: 'dismissNotice',
		value: function dismissNotice(notice) {
			return this.components.adminNotice.dismiss(notice);
		}
	}]);

	return WPAdmin;
}(_page2.default);

exports.default = WPAdmin;
module.exports = exports['default'];
