// wp-e2e-page-objects/lib/pages/wp-admin/wp-admin-plugin-install.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _componentPluginInstallFilter = require('../../components/wp-admin/component-plugin-install-filter');

var _componentPluginInstallFilter2 = _interopRequireDefault(_componentPluginInstallFilter);

var _componentPluginCard = require('../../components/wp-admin/component-plugin-card');

var _componentPluginCard2 = _interopRequireDefault(_componentPluginCard);

var _wpAdmin = require('./wp-admin');

var _wpAdmin2 = _interopRequireDefault(_wpAdmin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Internal Dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var components = {
	filter: _componentPluginInstallFilter2.default
};

var defaultArgs = {
	url: '',
	visit: true,
	components: components
};

var WPAdminPluginInstall = function (_WPAdmin) {
	_inherits(WPAdminPluginInstall, _WPAdmin);

	function WPAdminPluginInstall(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdminPluginInstall);

		args = Object.assign(defaultArgs, args);
		return _possibleConstructorReturn(this, (WPAdminPluginInstall.__proto__ || Object.getPrototypeOf(WPAdminPluginInstall)).call(this, driver, args));
	}

	_createClass(WPAdminPluginInstall, [{
		key: 'search',
		value: function search(pluginSlug) {
			return this.components.filter.search(pluginSlug);
		}
	}, {
		key: 'install',
		value: function install(pluginSlug) {
			var pluginCard = new _componentPluginCard2.default(this.driver, pluginSlug);
			return pluginCard.install();
		}
	}, {
		key: 'activate',
		value: function activate(pluginSlug) {
			var pluginCard = new _componentPluginCard2.default(this.driver, pluginSlug);
			return pluginCard.activate();
		}
	}]);

	return WPAdminPluginInstall;
}(_wpAdmin2.default);

exports.default = WPAdminPluginInstall;
module.exports = exports['default'];
