// wp-e2e-page-objects/lib/pages/wp-admin/wp-admin-plugins.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _wpAdmin = require('./wp-admin');

var _wpAdmin2 = _interopRequireDefault(_wpAdmin);

var _wpAdminPluginInstall = require('./wp-admin-plugin-install');

var _wpAdminPluginInstall2 = _interopRequireDefault(_wpAdminPluginInstall);

var _componentPluginsList = require('../../components/wp-admin/component-plugins-list');

var _componentPluginsList2 = _interopRequireDefault(_componentPluginsList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * External dependencies
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


/**
 * Internal dependencies
 */


var ADD_NEW_SELECTOR = _seleniumWebdriver.By.css('h1 .page-title-action');

var components = {
	pluginsList: _componentPluginsList2.default
};

var defaultArgs = {
	url: '',
	visit: true,
	components: components
};

var WPAdminPlugins = function (_WPAdmin) {
	_inherits(WPAdminPlugins, _WPAdmin);

	function WPAdminPlugins(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdminPlugins);

		args = Object.assign(defaultArgs, args);
		return _possibleConstructorReturn(this, (WPAdminPlugins.__proto__ || Object.getPrototypeOf(WPAdminPlugins)).call(this, driver, args));
	}

	_createClass(WPAdminPlugins, [{
		key: 'add',
		value: function add() {
			_wpE2eWebdriver.WebDriverHelper.clickWhenClickable(this.driver, ADD_NEW_SELECTOR);
			return new _wpAdminPluginInstall2.default(this.driver, { visit: false });
		}
	}, {
		key: 'activate',
		value: function activate(pluginSlug) {
			return this.components.pluginsList.activate(pluginSlug);
		}
	}, {
		key: 'deactivate',
		value: function deactivate(pluginSlug) {
			return this.components.pluginsList.deactivate(pluginSlug);
		}
	}, {
		key: 'delete',
		value: function _delete(pluginSlug) {
			return this.components.pluginsList.delete(pluginSlug);
		}
	}]);

	return WPAdminPlugins;
}(_wpAdmin2.default);

exports.default = WPAdminPlugins;
module.exports = exports['default'];
