// wc-e2e-page-objects/lib/pages/wp-admin/wp-admin-wc-settings-products-general.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _seleniumWebdriver = require('selenium-webdriver');

var _wpE2eWebdriver = require('wp-e2e-webdriver');

var _helper = require('../../helper');

var wcHelper = _interopRequireWildcard(_helper);

var _wpAdminWcSettings = require('./wp-admin-wc-settings');

var _wpAdminWcSettings2 = _interopRequireDefault(_wpAdminWcSettings);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * @module WPAdminWCSettingsProductsGeneral
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

/**
 * External dependencies
 */


/**
 * Internal Dependencies
 */


var WEIGHT_UNIT_SELECTOR = _seleniumWebdriver.By.css('#s2id_woocommerce_weight_unit .select2-choice b');
var DIMENSIONS_UNIT_SELECTOR = _seleniumWebdriver.By.css('#s2id_woocommerce_dimension_unit .select2-choice b');
var ENABLE_REVIEW_RATING_SELECTOR = _seleniumWebdriver.By.css('#woocommerce_enable_review_rating');
var REVIEW_RATING_REQUIRED_SELECTOR = _seleniumWebdriver.By.css('#woocommerce_review_rating_required');
var REVIEW_RATING_VERIFICATION_LABEL_SELECTOR = _seleniumWebdriver.By.css('#woocommerce_review_rating_verification_label');
var REVIEW_RATING_VERIFICATION_REQUIRED_SELECTOR = _seleniumWebdriver.By.css('#woocommerce_review_rating_verification_required');

var defaultArgs = {
	url: '',
	visit: true
};

/**
 * The Products: General settings screen
 *
 * @extends WPAdminWCSettings
 */

var WPAdminWCSettingsProductsGeneral = function (_WPAdminWCSettings) {
	_inherits(WPAdminWCSettingsProductsGeneral, _WPAdminWCSettings);

	/**
 	* @param {WebDriver} driver   - Instance of WebDriver.
 	* @param {object}    args     - Configuration arguments.
 */
	function WPAdminWCSettingsProductsGeneral(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdminWCSettingsProductsGeneral);

		args = Object.assign(defaultArgs, args);
		return _possibleConstructorReturn(this, (WPAdminWCSettingsProductsGeneral.__proto__ || Object.getPrototypeOf(WPAdminWCSettingsProductsGeneral)).call(this, driver, args));
	}

	/**
 * Select the weight unit.
 *
 	* @param  {string}    option - Weight unit text.
 	* @return {Promise}   Promise that evaluates to `true` if weight unit selected successfully, `false` otherwise.
 */


	_createClass(WPAdminWCSettingsProductsGeneral, [{
		key: 'selectWeightUnit',
		value: function selectWeightUnit(option) {
			return wcHelper.select2Option(this.driver, WEIGHT_UNIT_SELECTOR, option);
		}

		/**
  * Select the dimensions unit.
  *
  	* @param  {string}    option - Dimensions unit text.
  	* @return {Promise}   Promise that evaluates to `true` if dimensions unit selected successfully, `false` otherwise.
  */

	}, {
		key: 'selectDimensionsUnit',
		value: function selectDimensionsUnit(option) {
			return wcHelper.select2Option(this.driver, DIMENSIONS_UNIT_SELECTOR, option);
		}

		/**
  * Check the "Enable ratings on reviews" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets checked successfully, `false` otherwise.
  */

	}, {
		key: 'checkEnableRatingsOnReviews',
		value: function checkEnableRatingsOnReviews() {
			_wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, ENABLE_REVIEW_RATING_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, ENABLE_REVIEW_RATING_SELECTOR);
		}

		/**
  * Uncheck the "Enable ratings on reviews" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets unchecked successfully, `false` otherwise.
  */

	}, {
		key: 'uncheckEnableRatingsOnReviews',
		value: function uncheckEnableRatingsOnReviews() {
			_wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, ENABLE_REVIEW_RATING_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, ENABLE_REVIEW_RATING_SELECTOR);
		}

		/**
  * Check the "Ratings are required to leave a review" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets checked successfully, `false` otherwise.
  */

	}, {
		key: 'checkRatingsAreRequiredToLeaveReview',
		value: function checkRatingsAreRequiredToLeaveReview() {
			_wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_REQUIRED_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_REQUIRED_SELECTOR);
		}

		/**
  * Uncheck the "Ratings are required to leave a review" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets unchecked successfully, `false` otherwise.
  */

	}, {
		key: 'uncheckRatingsAreRequiredToLeaveReview',
		value: function uncheckRatingsAreRequiredToLeaveReview() {
			_wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_REQUIRED_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_REQUIRED_SELECTOR);
		}

		/**
  * Check the "Show 'verified owner' label for customer reviews" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets checked successfully, `false` otherwise.
  */

	}, {
		key: 'checkShowVerifiedOwnerLabel',
		value: function checkShowVerifiedOwnerLabel() {
			_wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_VERIFICATION_LABEL_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_VERIFICATION_LABEL_SELECTOR);
		}

		/**
  * Uncheck the "Show 'verified owner' label for customer reviews" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets unchecked successfully, `false` otherwise.
  */

	}, {
		key: 'uncheckShowVerifiedOwnerLabel',
		value: function uncheckShowVerifiedOwnerLabel() {
			_wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_VERIFICATION_LABEL_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_VERIFICATION_LABEL_SELECTOR);
		}

		/**
  * Check the "Only allow reviews from 'verified owners'" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets checked successfully, `false` otherwise.
  */

	}, {
		key: 'checkOnlyAllowReviewsFromVerifiedOwners',
		value: function checkOnlyAllowReviewsFromVerifiedOwners() {
			_wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_VERIFICATION_REQUIRED_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_VERIFICATION_REQUIRED_SELECTOR);
		}

		/**
  * Uncheck the "Only allow reviews from 'verified owners'" checkbox.
  *
  	* @return {Promise}   Promise that evaluates to `true` if box is/gets unchecked successfully, `false` otherwise.
  */

	}, {
		key: 'uncheckOnlyAllowReviewsFromVerifiedOwners',
		value: function uncheckOnlyAllowReviewsFromVerifiedOwners() {
			_wpE2eWebdriver.WebDriverHelper.setCheckbox(this.driver, REVIEW_RATING_VERIFICATION_REQUIRED_SELECTOR);
			return _wpE2eWebdriver.WebDriverHelper.unsetCheckbox(this.driver, REVIEW_RATING_VERIFICATION_REQUIRED_SELECTOR);
		}
	}]);

	return WPAdminWCSettingsProductsGeneral;
}(_wpAdminWcSettings2.default);

exports.default = WPAdminWCSettingsProductsGeneral;
module.exports = exports['default'];
