// wc-e2e-page-objects/lib/pages/wp-admin/wp-admin-order-edit.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _wpE2ePageObjects = require('wp-e2e-page-objects');

var _componentMetaBoxOrderData = require('../../components/wp-admin/component-meta-box-order-data');

var _componentMetaBoxOrderData2 = _interopRequireDefault(_componentMetaBoxOrderData);

var _componentMetaBoxOrderItems = require('../../components/wp-admin/component-meta-box-order-items');

var _componentMetaBoxOrderItems2 = _interopRequireDefault(_componentMetaBoxOrderItems);

var _componentMetaBoxOrderActions = require('../../components/wp-admin/component-meta-box-order-actions');

var _componentMetaBoxOrderActions2 = _interopRequireDefault(_componentMetaBoxOrderActions);

var _componentMetaBoxOrderNotes = require('../../components/wp-admin/component-meta-box-order-notes');

var _componentMetaBoxOrderNotes2 = _interopRequireDefault(_componentMetaBoxOrderNotes);

var _componentMetaBoxOrderDownloads = require('../../components/wp-admin/component-meta-box-order-downloads');

var _componentMetaBoxOrderDownloads2 = _interopRequireDefault(_componentMetaBoxOrderDownloads);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * @module WPAdminOrderEdit
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

/**
 * External dependencies
 */


/**
 * Internal dependencies
 */


/**
 * The admin Edit Order page
 *
 * @extends WPAdminPostEdit
 */
var WPAdminOrderEdit = function (_WPAdminPostEdit) {
	_inherits(WPAdminOrderEdit, _WPAdminPostEdit);

	/**
 	* @param {WebDriver} driver   - Instance of WebDriver.
 	* @param {object}    args     - Configuration arguments.
 */
	function WPAdminOrderEdit(driver) {
		var args = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

		_classCallCheck(this, WPAdminOrderEdit);

		args = Object.assign({
			components: {
				metaBoxOrderData: _componentMetaBoxOrderData2.default,
				metaBoxOrderItems: _componentMetaBoxOrderItems2.default,
				metaBoxOrderActions: _componentMetaBoxOrderActions2.default,
				metaBoxOrderNotes: _componentMetaBoxOrderNotes2.default,
				metaBoxDownloadableProductPermissions: _componentMetaBoxOrderDownloads2.default
			}
		}, args);
		return _possibleConstructorReturn(this, (WPAdminOrderEdit.__proto__ || Object.getPrototypeOf(WPAdminOrderEdit)).call(this, driver, args));
	}

	return WPAdminOrderEdit;
}(_wpE2ePageObjects.WPAdminPostEdit);

exports.default = WPAdminOrderEdit;
module.exports = exports['default'];
